view/frontend/web/js/view/payment/method-renderer/base.js

```diff
@@ -0,0 +1,409 @@
+            const acquiredUpdateSessionUrl = urlBuilder.createUrl(
+                '/acquired/session/update/' + this.nonce + "/" + sessionId,
+                {}
+            );
+
+            await storage.post(acquiredUpdateSessionUrl, {});
+        },
+
+        /**
+         * Get confirm parameters
+         *
+         * @returns {Promise<void>}
+         */
+        getConfirmParams: async function () {
+            let getConfirmParamsUrl = urlBuilder.createUrl('/acquired/confirm-params/' + this.nonce, {});
+            try {
+                let confirmParams = await storage.post(getConfirmParamsUrl);
+                if (_.indexOf(confirmParams, 0) !== -1 || confirmParams.length === 1) {
+                    return confirmParams[0];
+                } else {
+                    return {};
+                }
+            } catch (error) {
+                this.nonce = this.generateNonce();
+                throw new Error($t('There was an issue confirming the payment.'));
+            }
+        },
+
+        /**
+         * Get data
+         * @returns {Object}
+         */
+        getData: function () {
+            return {
+                'method': this.getCode(),
+                'additional_data': {
+                    'transaction_id': this.transactionId(),
+                    'order_id': this.orderId(),
+                    'timestamp': this.timestamp()
+                }
+            };
+        },
+
+        /**
+         * Action to place order
+         * @param {String} key
+         */
+        placeOrder: async function (key) {
+            let self = this;
+
+            fullScreenLoader.startLoader();
+            this.isPlaceOrderActionAllowed(false);
+
+            try {
+                await this.updateSession();
+            } catch (error) {
+                messageList.addErrorMessage({
+                    message: error.message || $t('Payment session expired, please re-enter the payment data.')
+                });
+                await this.initAcquired(true);
+                this.resetPlaceOrder();
+                return false;
+            }
+
+            try {
+                await this.formSubmitListener();
+
+                if (key) {
+                    this.getPlaceOrderDeferredObject()
+                        .done(
+                            function () {
+                                self.afterPlaceOrder();
+                                if (self.redirectAfterPlaceOrder) {
+                                    redirectOnSuccessAction.execute();
+                                }
+                            }
+                        ).fail(
+                            function () {
+                                self.afterPlaceOrder();
+                                self.resetPlaceOrder();
+                            }
+                        ).always(
+                            function () {
+                                self.afterPlaceOrder();
```

Comment:
Does it matter that `self.afterPlaceOrder()` is going to trigger twice?

- [ ] Validate `self.afterPlaceOrder()` duplication.

Reply:
afterPlaceOrder is empty, but I agree we can remove the always part